### PR TITLE
[RW-8400][risk=low] Add legacyAccountId and count that as free tier account. 

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
This should not affect the existing logic as it is the same value as AccountId. 

The unaffected field without using this new value: 
1: freeTierBillingAccountId from the UI usage: https://cs.github.com/?scope=org%3Aall-of-us&scopeName=all-of-us&q=freeTierBillingAccountId
2: freeTierBillingAccountName in the workspaceController, workspaceServiceImpl when creating/updaing workspaces. 
Once those should always do with new billing account
3: exportBigQueryTable. Still the same value, but we need to update sql in BigQuery view

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java [hunk 1]
```diff
@@ -70,10 +70,18 @@ public static class BillingConfig {
...
 
+    // The legacy free tier billing account id that is migrating away. This value helps to make
+    // migration process smooth.
+    public String legacyAccountId;
+
     public String freeTierBillingAccountName() {
       return "billingAccounts/" + accountId;
     }
 
+    public String legacyFreeTierBillingAccountName() {
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:81]: nit: perhaps add a helper here like `Set<String> freeTierBillingAccountNames()`

should make the two comparison code sites much easier

**Review by calbach — COMMENTED** [review]
I'm on the fence on updating workspaceserviceimpl:

https://cs.github.com/all-of-us/workbench/blob/9b4a5fd6773ea9fa9ee79b97e280f83120d643db/api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java?q=freeTierBillingAccountName#L397

If we need to revert the migration for some reason, it could be useful to allow either account. Also, people who are using the app in the middle of our migration might be affected - but this would be a very small number of users. I don't feel strongly.

File: api/config/config_preprod.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "gceVmZone": "us-central1-a"
   },
   "billing": {
+    "legacyAccountId": "00293C-5DEA2D-6887E7",
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/config/config_preprod.json:23]: Yes, in my design doc, I mentioned that we can keep this field for future migration. But I throught I would leave it the same as accountId. Good idea to allow this empty.

[the remark this answers is not in the document]
Comment (calbach) [reply api/config/config_preprod.json:23]: Did you mean to make this change? was expecting the value to be empty or missing here, and then `legacyFreeTierBillingAccountName` returns an optional

File: api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java [hunk 3]
```diff
@@ -70,10 +72,23 @@ public static class BillingConfig {
...
+      return "billingAccounts/" + legacyAccountId;
+    }
+
+    /// All valid free tier billing accounts, including accountId and legacyAccountId.
+    public Set<String> freeTierBillingAccountNames() {
+      return ImmutableSet.of("billingAccounts/" + accountId, "billingAccounts/" + legacyAccountId);
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/config/WorkbenchConfig.java:89]: nit: probably better that this calls the helpers `of(freetierBilingAccountName(), legacyFreeTierBillingAccountName())`

**Review by calbach — APPROVED** [review]

**Pull request merged**